src/web/views/Content/Select/index.jsx

```diff
@@ -0,0 +1,130 @@
+      this.props.fetchContentCategories()
+      this.setState({ show: true, callback })
+    }
+  }
+
+  componentWillUnmount() {
+    delete window.botpress.pickContent
+  }
+
+  search = event => {
+    this.setState({ searchTerm: event.target.value })
+    this.props.fetchContentMessagesRecent({ searchTerm: event.target.value })
+  }
+
+  handleCreate = () => {
+    this.props
+      .upsertContentMessages({ categoryId: this.state.category.id, formData: this.state.contentToEdit })
+      .then(() =>
+        Promise.all([
+          this.props.fetchContentMessagesRecent({ searchTerm: this.state.searchTerm }),
+          this.props.fetchContentMessagesCount()
+        ])
+      )
+      .then(() => this.setState({ category: null, contentToEdit: null }))
+  }
+
+  handlePick(item) {
+    this.setState({ show: false })
+    this.state.callback(item)
+  }
+
+  handleFormEdited = data => {
+    this.setState({ contentToEdit: data })
+  }
+
+  render() {
+    const props = this.props
+    const noop = () => {}
+
+    const onClose = () => this.setState({ show: false })
+    const onSubmit = () => this.setState({ show: false })
+
+    const schema = (this.state.category || {}).schema || { json: {}, ui: {} }
+
+    return (
+      <Modal animation={false} show={this.state.show} onHide={onClose} container={document.getElementById('app')}>
+        <Modal.Header closeButton>
+          <Modal.Title>Pick Content</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <input
+            type="text"
+            className="form-control"
+            placeholder={`Search all content elements (${this.props.messagesCount})`}
+            aria-label="Search content elements"
+            onKeyUp={this.search}
+          />
+          <hr />
+          <div className="list-group">
+            {this.props.categories.map(category => (
+              <a
+                href="#"
+                onClick={() => this.setState({ category, contentToEdit: {} })}
+                className={`list-group-item list-group-item-action ${classnames(style.createItem)}`}
+              >
+                Create new {category.title}
+              </a>
+            ))}
+            {this.props.messages.map(message => (
+              <a href="#" className="list-group-item list-group-item-action" onClick={() => this.handlePick(message)}>
+                {`[${message.categoryId}] ${message.data && message.data.question}`}
+              </a>
+            ))}
+          </div>
+        </Modal.Body>
+
+        <CreateModal
+          show={Boolean(this.state.category)}
+          schema={schema.json}
+          uiSchema={schema.ui}
+          handleClose={() => this.setState({ category: null })}
+          formData={this.state.contentToEdit}
+          handleEdit={this.handleFormEdited}
+          handleCreateOrUpdate={this.handleCreate}
+        />
+      </Modal>
+    )
+  }
+}
+
+const mapStateToProps = state => ({
+  messages: state.content.recentMessages,
```

Comment:
I see everywhere `messages`, let's rename that to `elements` since a content element is not necessarily a message. In the case of Flows, it's always messages but content can represent anything

Reply:
@slvnperron , they are also sometimes called items. For consistency, can I call them `contentItems` everywhere?

Reply:
Yep